src/imagery/i.eodag/i.eodag.py

```diff
@@ -813,15 +813,47 @@ def products_compare(first, second):
         for sort_key in sort_keys:
             if sort_key == "ingestiondate":
                 if "startTimeFromAscendingNode" not in first.properties:
+                    gs.warning(
+                        _("Could not sort by ingestion date, please report this issue")
+                    )
+                    continue
+                if "startTimeFromAscendingNode" not in second.properties:
+                    gs.warning(
+                        _("Could not sort by ingestion date, please report this issue")
+                    )
                     continue
                 first_value = first.properties["startTimeFromAscendingNode"]
                 second_value = second.properties["startTimeFromAscendingNode"]
             elif sort_key == "cloudcover":
                 if "cloudCover" not in first.properties:
-                    continue
-                first_value = first.properties["cloudCover"]
-                second_value = second.properties["cloudCover"]
+                    gs.warning(
+                        _(
+                            "Cloud cover not available for scene {}".format(
+                                first.properties.get("title", "NA")
+                            )
+                        )
+                    )
+                if "cloudCover" not in second.properties:
+                    gs.warning(
+                        _(
+                            "Cloud cover not available for scene {}".format(
+                                second.properties.get("title", "NA")
+                            )
+                        )
+                    )
+                first_value = first.properties.get("cloudCover", int(1e9))
+                second_value = second.properties.get("cloudCover", int(1e9))
             elif sort_key == "footprint":
+                if "title" not in first.properties:
+                    gs.warning(
+                        _("Could not sort by footprint, please report this issue")
+                    )
+                    continue
+                if "title" not in second.properties:
+                    gs.warning(
+                        _("Could not sort by footprint, please report this issue")
+                    )
+                    continue
```

Comment:
Here (as well as above) if conditions can be merged.

```suggestion
                if "title" not in first.properties or "title" not in second.properties:
                    gs.warning(
                        _("Could not sort by footprint, please report this issue")
                    )
                    continue
```

A question is: how or where to "report this issue"? Maybe better to have a similar warning message as for cloudcover or the like...